Userland/Libraries/LibTLS/TLSv12.cpp

```diff
@@ -4,622 +4,167 @@
-        auto hash = hasher.digest();
-
-        Crypto::Curves::SECP256r1 curve;
-        auto result = curve.verify(hash.bytes(), issuer.public_key.raw_key, subject.signature_value);
-        if (result.is_error()) {
-            dbgln("verify_certificate_pair: Failed to check SECP256r1 signature {}", result.release_error());
-            return false;
-        }
-        return result.value();
-    }
-    case SupportedGroup::SECP384R1: {
-        Crypto::Hash::Manager hasher(kind);
-        hasher.update(subject.tbs_asn1.bytes());
-        auto hash = hasher.digest();
-
-        Crypto::Curves::SECP384r1 curve;
-        auto result = curve.verify(hash.bytes(), issuer.public_key.raw_key, subject.signature_value);
-        if (result.is_error()) {
-            dbgln("verify_certificate_pair: Failed to check SECP384r1 signature {}", result.release_error());
-            return false;
-        }
-        return result.value();
-    }
-    case SupportedGroup::X25519: {
-        Crypto::Curves::Ed25519 curve;
-        auto result = curve.verify(issuer.public_key.raw_key, subject.signature_value, subject.tbs_asn1.bytes());
-        if (!result) {
-            dbgln("verify_certificate_pair: Failed to check Ed25519 signature");
-            return false;
-        }
-        return result;
-    }
-    default:
-        dbgln("verify_certificate_pair: Don't know how to verify signature for curve {}", to_underlying(issuer.public_key.algorithm.ec_parameters));
-        return false;
-    }
+    auto result = wolfSSL_write(ssl, bytes.data(), bytes.size());
+    if (result < 0)
+        return Error::from_string_view(error_text(ssl, result));
+    return static_cast<size_t>(result);
 }
 
-template<typename HMACType>
-static void hmac_pseudorandom_function(Bytes output, ReadonlyBytes secret, u8 const* label, size_t label_length, ReadonlyBytes seed, ReadonlyBytes seed_b)
+bool WolfTLS::is_eof() const
 {
-    if (!secret.size()) {
-        dbgln("null secret");
-        return;
-    }
-
-    auto append_label_seed = [&](auto& hmac) {
-        hmac.update(label, label_length);
-        hmac.update(seed);
-        if (seed_b.size() > 0)
-            hmac.update(seed_b);
-    };
-
-    HMACType hmac(secret);
-    append_label_seed(hmac);
-
-    constexpr auto digest_size = hmac.digest_size();
-    u8 digest[digest_size];
-    auto digest_0 = Bytes { digest, digest_size };
-
-    digest_0.overwrite(0, hmac.digest().immutable_data(), digest_size);
-
-    size_t index = 0;
-    while (index < output.size()) {
-        hmac.update(digest_0);
-        append_label_seed(hmac);
-        auto digest_1 = hmac.digest();
-
-        auto copy_size = min(digest_size, output.size() - index);
-
-        output.overwrite(index, digest_1.immutable_data(), copy_size);
-        index += copy_size;
-
-        digest_0.overwrite(0, hmac.process(digest_0).immutable_data(), digest_size);
-    }
+    return wolfSSL_get_error(ssl, 0) == SSL_ERROR_WANT_READ;
```

Comment:
Is that the right way to detect eof? SSL_ERROR_WANT_READ seems like just EWOULDBLOCK, which doesn't look to match the previous behavior

Reply:
tbh that's the only thing I managed to come up with that "worked".
I'll rewrite it as "no buffer and underlying closed", hopefully that'll work.

Reply:
seems fine 🤷